### PR TITLE
Convert more `ObjectDisposedExceptions` to use `ThrowIf`

### LINKED ISSUE #8184: Use `ObjectDisposedException.ThrowIf` in more places
### Is your feature request related to a problem? Please describe

`ObjectDisposedException.ThrowIf` was introduced in .NET7, we should make use of it in more places.

https://learn.microsoft.com/en-us/dotnet/api/system.objectdisposedexception.throwif?view=net-7.0

### Describe the solution you'd like and alternatives you've considered

n/a

### Will this feature affect UI controls?

n/a

### PR DESCRIPTION
**Pull request by elachlan**
Fixes #8184

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/8211)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/GridEntry.cs [hunk 1]
```diff
@@ -506,11 +506,7 @@ public override GridItemCollection GridItems
         {
             get
             {
-                if (Disposed)
-                {
-                    throw new ObjectDisposedException(SR.GridItemDisposed);
```
[the remark this answers is not in the document]
Comment (elachlan) [reply src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/GridEntry.cs:511]: done.

**Comment by dreddy-work** [comment]
@JeremyKuhne / @Tanya-Solyanik is it possible that anyone could take dependency on Exception messages?

**Review by dreddy-work — APPROVED** [review]
LGTM.

**Comment by RussKie** [comment]
> is it possible that anyone could take dependency on Exception messages?

It's totally possible. Developers are a fascinating bunch, and they find uncountable way to use and abuse the API we provide.
A better question - is it something we really care about? The answer is likely be in the vicinity of "no, no really". My 2c.

**Comment by Tanya-Solyanik** [comment]
@dreddy-work  - this is a `reasonable grey area` per [runtime breaking changes guidance](https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/breaking-changes.md). If you think this message is important, we can extend `ThrowIf` to support a custom message. I feel that `ObjectDisposed` is an unexpected exception that customers are unlikely to depend on and am ok with this change. It's not a "something not found" exception that suggests the developer to search elsewhere.  We might be breaking tests that are comparing logs in the error scenario.

**Comment by dreddy-work** [comment]
Thank you @Tanya-Solyanik. Lets see if we hear any report. We would be removing more and more messages with similar changes in this repo, but we can try group them into early previews and respond on feedback.

**Pull request merged**